test/e2e/tools/eks-anywhere-test-tool/pkg/cloudwatch/cloudwatch.go

```diff
@@ -83,5 +84,20 @@ func (c Cloudwatch) getLogSegment(logGroupName string, logStreamName string, sta
 	if endTime != nil {
 		input.EndTime = endTime
 	}
-	return c.svc.GetLogEvents(input)
+	logEvents, err := c.svc.GetLogEvents(input)
+	if err != nil {
+		if isInvalidParameterError(err) {
+			logger.Info("log stream does not exist. Proceeding to fetch next log events", "logStream", logStreamName)
+		} else {
```

Comment:
can we remove this `else` block and just return `logEvents, err`

Reply:
that would still return the `InvalidParameterException` error. However, I did just update this to only check if the error is the `InvalidParameterException` error. If so, give the warning & return `logEvents, nil`. All other cases will return `logEvents, err`. That simplifies it a bit more.